File: 1.two-sum.cpp
/*
 * @lc app=leetcode id=1 lang=cpp
 *
 * [1] Two Sum
 */
#include <vector>
#include <unordered_map>
using namespace std;
// @lc code=start
class Solution
{
public:
    vector<int> twoSum(vector<int> &nums, int target)
    {
        int len = nums.size();
        unordered_map<int, short> umap;
        for (int i = 0; i != len; ++i)
        {
            if (umap.count(nums[i]))
            {
                return vector<int>{umap[nums[i]], i};
            }
            else
            {
                umap[target - nums[i]] = i;
            }
        }
        return vector<int>();
    }
};
// @lc code=end

File: 1.two-sum.java
import java.util.HashMap;
import java.util.Map;

/*
 * @lc app=leetcode id=1 lang=java
 *
 * [1] Two Sum
 */

// @lc code=start
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int len = nums.length;
        Map<Integer, Short> umap = new HashMap<>();
        for (short i = 0; i != len; ++i) {
            if (umap.containsKey(nums[i])) {

                return new int[] { umap.get(nums[i]).intValue(), i };
            } else {
                umap.put((target - nums[i]), i);
            }
        }
        return null;
    }
}
// @lc code=end
